portfolio/src/main/webapp/charts.js

```diff
@@ -0,0 +1,215 @@
+ */
+function createChartBySex(covidData) {
+  const data = new google.visualization.DataTable();
+  data.addColumn('string', 'Sex');
+  data.addColumn('number', 'Covid-19 Cases');
+
+  data.addRows([
+    ['Women', covidData['M']],
+    ['Men', covidData['H']]
+  ]);
+
+  const options = {
+    chartArea: {
+      height: '80%',
+      width: '80%'
+    },
+    legend: {alignment: 'center', position: 'bottom'},
+    title: 'Covid-19 Cases: Women and Men',
+    pieHole: 0.4,
+    width: 500,
+    height: 400
+  };
+
+  const chart = new google.visualization.PieChart(
+    document.getElementById('covidBySex-container'));
+  chart.draw(data, options);
+}
+
+/**
+ * Create a bar chart showing the no. of cases in
+ * each municipality. In descending order, from
+ * top to bottom
+ * @param {Object<string, number>} covidData 
+ */
+function createChartByMunicipality(covidData) {
+  const data = new google.visualization.DataTable();
+  data.addColumn('string', '');
+  data.addColumn('number', 'Covid-19 Cases');
+
+  Object.keys(covidData).forEach((municipality) => {
+    data.addRow([municipality, covidData[municipality]]);
+  });
+  
+  const options = {
+    axes: {
+      x: {
+        0: { 
+          side: 'top',
+          label: 'No. of confirmed cases',
+          range: {min: 0, max: covidData[Object.keys(covidData)[0]]}},
+      }
+    },
+    bars: 'horizontal',
+    chart: {
+      title: 'Cumulative Confirmed Covid-19 Cases (Municipality)'
+    },
+    colors: ['red'],
+    hAxis: {
+      baselineColor: '#fff',
+      format: 'decimal'
+    },
+    height: 1200,
+    legend: { position: 'none'},
+    width: 800
+  };
+
+  const chart = new google.charts.Bar(
+    document.getElementById('covidByMunicipality-container'));
+  chart.draw(data, google.charts.Bar.convertOptions(options));
+}
+
+/**
+ * Create a histogram with data the age of positive cases
+ * @param {Object<string, number>} covidData 
+ */
+function createChartByAgeRange(covidData) {
+  const data = new google.visualization.DataTable();
+  data.addColumn('string', '');
+  data.addColumn('number', 'Covid-19 Cases');
+
+  Object.keys(covidData).forEach((idEntry) => {
+    data.addRow([idEntry, covidData[idEntry]]);
+  });
+
+  const options = {
+    chartArea: {
+      height: '80%',
+      width: '80%'
+    },
+    colors: ['red'],
+    dataOpacity: .9,
+    hAxis: {
+      slantedText: true,
+      slantedTextAngle: 60,
+      showTextEvery: 1,
+      viewWindowMode: 'maximized'
+    },
+    histogram: { bucketSize: 5},
+    height: 400,
+    legend: { position: 'none'},
+    title: 'Cumulative Confirmed Covid-19 Cases (Age Range)',
+    vAxis: {
+      baselineColor: '#fff',
+      format: 'decimal',
+      viewWindowMode: 'maximized'
+    },
+    width: 800
+  };
+
+  const chart = new google.visualization.Histogram(
+      document.getElementById('covidByAgeRange-container'));
+  chart.draw(data, options);
+}
+
+function createChartByState(covidData) {
```

Comment:
As I'm reading these more closely, they seem repetitive enough that you might be able to make one generic method that handles creating any given chart type, it just takes in a few parameters depending on chart type?

Reply:
I was actually thinking the same, there is definitively some logic I can reuse and pass some arguments to a generic method.